src/core/ddsi/src/q_receive.c

```diff
@@ -2321,7 +2321,7 @@ static void handle_regular (struct receiver_state *rst, ddsrt_etime_t tnow, stru
     lease_renew (lease, tnow);
   if ((lease = ddsrt_atomic_ldvoidp (&pwr->c.proxypp->minl_man)) != NULL && renew_manbypp_lease)
     lease_renew (lease, tnow);
-  if (pwr->lease && pwr->c.xqos->liveliness.kind == DDS_LIVELINESS_MANUAL_BY_TOPIC)
+  if (pwr->c.xqos->liveliness.lease_duration != DDS_INFINITY && pwr->c.xqos->liveliness.kind == DDS_LIVELINESS_MANUAL_BY_TOPIC)
```

Comment:
Whether checking the lease duration (here in the QoS, elsewhere using the `lease_duration` field) against ∞ instead of the lease against a null pointer is clearer I don't know ... In any case, it is not going to make a difference. This is better for consistency, but at the same time, it relies more heavily on keeping track of how the various fields relate to each other (see table) and that's a downside.